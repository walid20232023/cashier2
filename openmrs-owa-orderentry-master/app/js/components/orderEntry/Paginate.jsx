import React from 'react';
import PropTypes from 'prop-types';


const maxNumberOfOrdersOnAPage = 10;

const getCurrentPageNumber = (nextPageUrl, totalNumberOfOrders) => {
  if (nextPageUrl) {
    const strings = nextPageUrl.split('=');
    const pageNumber = Math.floor(strings[strings.length - 1] / maxNumberOfOrdersOnAPage);
    return pageNumber;
  }
  return Math.ceil(totalNumberOfOrders / maxNumberOfOrdersOnAPage);
};

const getNumberOfAllPages = totalNumberOfOrders =>
  Math.ceil(totalNumberOfOrders / maxNumberOfOrdersOnAPage);


const Paginate = ({
  totalPage,
  nextPageUrl,
  dispatch,
  fetchNew,
  prevPageUrl,
}) => (
  <div className="pagination">
    <div
      className={prevPageUrl ? "page-link prev" : "page-link-disabled"}
      value="previous"
      onClick={() => dispatch(fetchNew(prevPageUrl))}
      role="button"
      tabIndex={0}>
      <p>❮❮</p>
    </div>

    <div className="page-link">
      <p>{getCurrentPageNumber(nextPageUrl, totalPage)}
      </p>
    </div>
    <div className="page-link no-click">
      <p>/</p>
    </div>
    <div className="page-link">
      <p>
        <b>{getNumberOfAllPages(totalPage)}</b>
      </p>
    </div>
    <div
      className={nextPageUrl ? "page-link next" : "page-link-disabled"}
      value={nextPageUrl}
      onClick={() => dispatch(fetchNew(nextPageUrl))}
      role="button"
      tabIndex={0}>
      <p>❯❯</p>
    </div>
  </div>
);

export default Paginate;

Paginate.defaultProps = {
  totalPage: 10,
  nextPageUrl: '',
  prevPageUrl: '',
};

Paginate.propTypes = {
  totalPage: PropTypes.number,
  nextPageUrl: PropTypes.string,
  prevPageUrl: PropTypes.string,
  dispatch: PropTypes.func.isRequired,
  fetchNew: PropTypes.func.isRequired,
};
