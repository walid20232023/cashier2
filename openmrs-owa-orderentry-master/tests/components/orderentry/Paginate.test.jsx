import React from 'react';
import Paginate from '../../../app/js/components/orderEntry/Paginate';

const props = {
  totalPage: 193,
  nextPageUrl: 'http://next=page=some=random=page=and=the=length=20',
  dispatch: jest.fn(),
  prevPageUrl: 'http://prev=page=some=random=page=and=the=length=10',
  patientId: '123eeeeeer',
  fetchNew: jest.fn(),
}

const wrapper = shallow(<Paginate {...props} />)

describe('Pagination component test-suite', () => {
  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches action to fetch new data when the next button is clicked', () => {
    wrapper.find('.next').simulate('click');
    expect(props.fetchNew).toHaveBeenCalledWith(props.nextPageUrl);
  })

  it('dispatches action to fetch new data when the previous button is clicked', () => {
    wrapper.find('.prev').simulate('click');
    expect(props.fetchNew).toHaveBeenCalledWith(props.prevPageUrl);
  })

  it('Disable the button when no url is passed', () => {
    wrapper.setProps({
      ...wrapper.props,
      prevPageUrl: '',
      nextPageUrl: '',
    })
    expect(wrapper.find('.page-link-disabled')).toHaveLength(2);
  });

  
});
