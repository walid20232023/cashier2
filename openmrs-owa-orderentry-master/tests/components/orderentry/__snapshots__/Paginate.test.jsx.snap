// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component test-suite renders properly 1`] = `
<div
  className="pagination"
>
  <div
    className="page-link prev"
    onClick={[Function]}
    role="button"
    tabIndex={0}
    value="previous"
  >
    <p>
      ❮❮
    </p>
  </div>
  <div
    className="page-link"
  >
    <p>
      2
    </p>
  </div>
  <div
    className="page-link no-click"
  >
    <p>
      /
    </p>
  </div>
  <div
    className="page-link"
  >
    <p>
      <b>
        20
      </b>
    </p>
  </div>
  <div
    className="page-link next"
    onClick={[Function]}
    role="button"
    tabIndex={0}
    value="http://next=page=some=random=page=and=the=length=20"
  >
    <p>
      ❯❯
    </p>
  </div>
</div>
`;
